tests/CorsMiddlewareTest.php

```diff
@@ -94,7 +93,7 @@ public function testShouldAcceptWildcardSettings(): void
         $this->assertEquals("https://subdomain.example.com", $response->getHeaderLine("Access-Control-Allow-Origin"));
         $this->assertEquals("true", $response->getHeaderLine("Access-Control-Allow-Credentials"));
         $this->assertEquals("Origin", $response->getHeaderLine("Vary"));
-        $this->assertEquals("Authorization,Etag", $response->getHeaderLine("Access-Control-Expose-Headers"));
+        $this->assertEquals("Authorization, Etag", $response->getHeaderLine("Access-Control-Expose-Headers"));
```

Comment:
eShould not change tests because it is hard to know if there is a BC break or not. This test without space is intentional because apparently at least some versions of Edge don't handle the space there well. 

https://github.com/tuupola/cors-middleware/pull/44

Reply:
Hehe ok, thank you - wasn't aware of that.
The behavior changed upstream so maybe it would be best to create a PR in neomerx/cors-psr7 - but as you already have found out, it seems to be kinda unmaintained.
So I think I'll add some sort of mitigation...

Reply:
Ok, tbh I'm not very happy with that "solution" but the alternatives aren't  much better.
I thought about overwriting `Settings::setExposedHeaders()`, but it's not possible to access the internal state of Settings which would lead to different states within the inheritance chain. Best would be to apply the changes directly upstream...

Reply:
It seems to come from here:

https://github.com/neomerx/cors-psr7/blob/0f968523b0b8215540915464aed08c5c4a4bf8b5/src/Strategies/Settings.php#L608

Have to think a bit what would be the best thing to do. Maybe the Edge bug has been already fixed.

Reply:
Yepp, found it already. Thanks to MS, the deep links to the edge issues don't work anymore. I'll dig a bit...

Reply:
But why does not this code fix the problem? The `fixHeaders()` method was added to fix the header [neomerx/cors-psr7](https://github.com/neomerx/cors-psr7) gives back.

https://github.com/tuupola/cors-middleware/blob/master/src/CorsMiddleware.php#L255-L263

Reply:
After the update, $headers[CorsResponseHeaders::EXPOSE_HEADERS] is longer an array. https://github.com/neomerx/cors-psr7/blob/0f968523b0b8215540915464aed08c5c4a4bf8b5/src/Strategies/Settings.php#L597

Reply:
Ok, so `fixHeaders()` should be changed to work with the new format.

Reply:
You can actually do a normal push. GitHub now has the squash feature which I use for PRs.

Reply:
> You can actually do a normal push. GitHub now has the squash feature which I use for PRs.

No problem if you prefer that instead of force push (it's the workflow I'm used to).

Reply:
> You can actually do a normal push. GitHub now has the squash feature which I use for PRs.

No problem if you prefer that instead of force push (it's the workflow I'm used to).

Reply:
> You can actually do a normal push. GitHub now has the squash feature which I use for PRs.

No problem if you prefer that instead of force push (it's the workflow I'm used to).

Reply:
> You can actually do a normal push. GitHub now has the squash feature which I use for PRs.

No problem if you prefer that instead of force push (it's the workflow I'm used to).